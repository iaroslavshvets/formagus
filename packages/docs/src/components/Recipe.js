import React from 'react';

export const Recipe = ({sandboxId}) => (
  <>
    <iframe
      id="recipe"
      title="formagus recipe"
      src={`https://codesandbox.io/embed/${sandboxId}?fontsize=13&module=/src/ExampleForm.tsx&view=split`}
      style={{
        display: 'block',
        width: '100%',
        border: 0,
        height: '100vh',
      }}
      sandbox="allow-modals allow-forms allow-popups allow-scripts allow-same-origin"
    />
  </>
);
