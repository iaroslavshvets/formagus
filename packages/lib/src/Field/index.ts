export * from './FieldClass';
export * from './Field';
export * from './Field.types';
