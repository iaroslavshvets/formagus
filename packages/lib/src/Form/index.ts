export * from './Form';
export * from './Form.types';
export * from './useFormController';
