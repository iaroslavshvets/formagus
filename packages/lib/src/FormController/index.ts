export * from './FormController';
export * from './FormController.types';
