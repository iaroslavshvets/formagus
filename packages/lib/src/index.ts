export type {
  ValidationFunction,
  FieldAdapter,
  AdapterProps,
  FieldMeta,
  EqualityCheckFunction,
  FieldProps,
} from './Field';
export type {FormProps} from './Form';
export type {
  FormAPI,
  FormValidationErrors,
  FormValues,
  FieldValidationState,
  FormField,
  FormFieldMeta,
} from './FormController';

export {utils} from './FormController/utils';
export {FormController} from './FormController';
export {Field} from './Field';
export {Form} from './Form';
export {FormPart} from './FormPart';
export {useFormApi} from './useFormApi';
export {injectFormApi} from './injectFormApi';
