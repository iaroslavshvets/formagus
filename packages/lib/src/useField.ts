import {useContext} from 'react';
import {FieldContext} from './Field/FieldContext';
import type {FormagusProps} from './Field/Field.types';

export const useField = () => {
  return useContext(FieldContext) as FormagusProps;
};
