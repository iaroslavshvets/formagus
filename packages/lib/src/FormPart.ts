import {observer} from 'mobx-react';
import {useFormController} from './Form/useFormController';
import type {FormProps} from './Form/Form.types';

export const FormPart = observer((props: {children: FormProps['children']}) => {
  const controller = useFormController();

  return props.children(controller.API)


});

(FormPart as any).displayName = 'FormagusFormPart';
