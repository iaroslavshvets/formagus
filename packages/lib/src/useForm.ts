import {useContext} from 'react';
import {FormControllerContext} from './Form/FormControllerContext';
import type {FormController} from './createFormController/createFormController.types';

export const useForm = (props?: {controller?: FormController}) => {
  const controllerFromContext = useContext(FormControllerContext);

  return ((props?.controller || controllerFromContext) as FormController).API;
};
