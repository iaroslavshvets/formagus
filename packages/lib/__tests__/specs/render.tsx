import {cleanup, render} from '@testing-library/react';
import React from 'react';
import {TestForm} from '../components/TestForm';
import {createTestFormDriver} from '../components/TestForm.driver';
import {InputAdapter} from '../components/InputAdapter';
import {Field} from '../../src';

test('Render', () => {
  afterEach(() => {
    return cleanup();
  });

  const wrapper = render(
    <TestForm>
      <Field name={TestForm.FIELD_ONE_NAME} adapter={InputAdapter} />
      <Field name={TestForm.FIELD_TWO_NAME}>
        {(props) => {
          return <InputAdapter {...props} />;
        }}
      </Field>
    </TestForm>,
  ).container;
  const formDriver = createTestFormDriver({wrapper});

  expect(formDriver.get.serialized()).toMatchSnapshot();
});
