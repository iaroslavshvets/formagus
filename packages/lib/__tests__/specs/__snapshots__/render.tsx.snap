// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render should render 1`] = `
<div>
  <form
    data-hook="test-form"
    novalidate=""
  >
    <div
      data-hook="field_one_name"
    >
      <input
        data-hook="input-field_one_name"
        name="field_one_name"
        value=""
      />
      <div
        data-hook="meta-props"
      >
        <span
          data-hook="meta_errors"
        >
          null
        </span>
        <span
          data-hook="meta_initialValue"
        >
          undefined
        </span>
        <span
          data-hook="meta_isActive"
        >
          false
        </span>
        <span
          data-hook="meta_isDirty"
        >
          false
        </span>
        <span
          data-hook="meta_isTouched"
        >
          false
        </span>
        <span
          data-hook="meta_isChanged"
        >
          false
        </span>
        <span
          data-hook="meta_isValidating"
        >
          false
        </span>
        <span
          data-hook="meta_form:isSubmitting"
        >
          false
        </span>
        <span
          data-hook="meta_form:isValidating"
        >
          false
        </span>
        <span
          data-hook="meta_form:isValid"
        >
          true
        </span>
        <span
          data-hook="meta_form:isDirty"
        >
          false
        </span>
        <span
          data-hook="meta_form:isTouched"
        >
          false
        </span>
        <span
          data-hook="meta_form:isChanged"
        >
          false
        </span>
        <span
          data-hook="meta_form:submitCount"
        >
          0
        </span>
      </div>
      <span
        data-hook="set-custom-state"
      />
      <span
        data-hook="callback"
      />
      <span
        data-hook="validate"
      />
    </div>
    <div
      data-hook="field_two_name"
    >
      <input
        data-hook="input-field_two_name"
        name="field_two_name"
        value=""
      />
      <div
        data-hook="meta-props"
      >
        <span
          data-hook="meta_errors"
        >
          null
        </span>
        <span
          data-hook="meta_initialValue"
        >
          undefined
        </span>
        <span
          data-hook="meta_isActive"
        >
          false
        </span>
        <span
          data-hook="meta_isDirty"
        >
          false
        </span>
        <span
          data-hook="meta_isTouched"
        >
          false
        </span>
        <span
          data-hook="meta_isChanged"
        >
          false
        </span>
        <span
          data-hook="meta_isValidating"
        >
          false
        </span>
        <span
          data-hook="meta_form:isSubmitting"
        >
          false
        </span>
        <span
          data-hook="meta_form:isValidating"
        >
          false
        </span>
        <span
          data-hook="meta_form:isValid"
        >
          true
        </span>
        <span
          data-hook="meta_form:isDirty"
        >
          false
        </span>
        <span
          data-hook="meta_form:isTouched"
        >
          false
        </span>
        <span
          data-hook="meta_form:isChanged"
        >
          false
        </span>
        <span
          data-hook="meta_form:submitCount"
        >
          0
        </span>
      </div>
      <span
        data-hook="set-custom-state"
      />
      <span
        data-hook="callback"
      />
      <span
        data-hook="validate"
      />
    </div>
    <div
      data-hook="form-values"
    >
      {}
    </div>
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
